src/webots/gui/WbSingleTaskApplication.cpp

```diff
@@ -130,9 +132,18 @@ void WbSingleTaskApplication::convertProto() const {
   // Generate a node structure
   WbNode::setInstantiateMode(true);
   WbNode *node = WbNode::regenerateProtoInstanceFromParameters(model, fields, true, "");
-  for (WbNode *subNode : node->subNodes(true))
-    if (dynamic_cast<WbBasicJoint *>(subNode))
+  for (WbNode *subNode : node->subNodes(true)) {
+    if (type == QString("URDF") && dynamic_cast<WbSolidReference *>(subNode))
```

Comment:
That is simpler and should work as well:
```suggestion
    if (type == "URDF" && dynamic_cast<WbSolidReference *>(subNode))
```